mllib/src/main/scala/org/apache/spark/mllib/tree/configuration/Strategy.scala

```diff
@@ -181,4 +181,17 @@ object Strategy {
       new Strategy(algo = Regression, impurity = Variance, maxDepth = 10,
         numClasses = 0)
   }
+
+  /**
+   * Construct a default set of parameters for [[org.apache.spark.mllib.tree.DecisionTree]]
+   * @param algo Algo.Classification or Algo.Regression
+   */
+  def defaultStategy(algo: Algo): Strategy = algo match {
```

Comment:
Same here. Let's update `defaultStrategy(algoStr: String)` to use the implementation here.